framework/test/unit/specs/modules/token/token_module.spec.ts

```diff
@@ -165,7 +266,100 @@ describe('token module', () => {
 			);
 		});
 	});
-	// TODO: Add #reducers test
-	// TODO: Add #afterBlockApply test
-	// TODO: Add #afterGenesisBlockApply test
+
+	describe('#afterBlockApply', () => {
+		const generatorPublicKey = getRandomBytes(20);
+		let block: any;
+		const minFee =
+			BigInt(genesisConfig.minFeePerByte) * BigInt(1) + BigInt(genesisConfig.baseFees[0].baseFee);
+		beforeEach(() => {
+			block = {
+				header: {
+					generatorPublicKey,
+					reward: BigInt(1),
+				},
+				payload: [
+					{
+						moduleType: 2,
+						assetType: 0,
+						getBytes: () => [1],
+						fee: BigInt(20000000),
+					},
+				],
+			};
+		});
+
+		describe('when block contains transactions', () => {
+			it('should update generator balance to give rewards and fees - minFee', async () => {
+				let expected = BigInt(senderAccount.token.balance) + block.header.reward;
+				for (const tx of block.payload) {
+					expected += tx.fee - minFee;
+				}
+				await tokenModule.afterBlockApply({
+					block,
+					stateStore,
+					reducerHandler,
+					consensus: {} as any,
+				});
+
+				expect(senderAccount.token.balance).toEqual(expected);
+			});
+
+			it('should update burntFee in the chain state', async () => {
+				const expected = BigInt(0) + minFee;
+				const expectedBuffer = Buffer.alloc(8);
+				expectedBuffer.writeBigInt64BE(expected);
+				await tokenModule.afterBlockApply({
+					block,
+					stateStore,
+					reducerHandler,
+					consensus: {} as any,
+				});
+
+				expect(stateStore.chain.set).toBeCalledWith(CHAIN_STATE_BURNT_FEE, expectedBuffer);
+			});
+		});
```

Comment:
Also, it's better to test where it includes transaction which has `baseFee`

Reply:
`baseFee` is included in the `minFee`